src/tests/JIT/HardwareIntrinsics/Arm/Shared/Helpers.cs

```diff
@@ -7744,12 +7744,12 @@ public static ulong getMaskUInt64()
 
         public static float getMaskSingle()
         {
-            return (float)(BitConverter.Int32BitsToSingle(TestLibrary.Generator.GetInt32()%(int)2));
+            return (float)(TestLibrary.Generator.GetUInt32()%(int)2);
```

Comment:
Since we're here, should we get rid of these casts, too?

```suggestion
            return (float)(TestLibrary.Generator.GetUInt32()%2);
```